pkg/workload/core.go

```diff
@@ -133,6 +133,42 @@ func createOperationGenerator(p *properties.Properties) *generator.Discrete {
 	return operationChooser
 }
 
+// Init --create all schemas for ycsb workload
+func (c *core) Init(db ycsb.DB) error {
+	// need to redesign the relation btw worklaod and db interface later.
+	sqlDB := db.ToSqlDB()
+	if sqlDB != nil {
+		tableName := c.p.GetString(prop.TableName, prop.TableNameDefault)
+		if c.p.GetBool(prop.DropData, prop.DropDataDefault) && !c.p.GetBool(prop.DoTransactions, true) {
+			if _, err := sqlDB.Exec(fmt.Sprintf("DROP TABLE IF EXISTS %s", tableName)); err != nil {
+				return err
+			}
+		}
+
+		fieldCount := c.p.GetInt64(prop.FieldCount, prop.FieldCountDefault)
+		fieldLength := c.p.GetInt64(prop.FieldLength, prop.FieldLengthDefault)
+
+		buf := new(bytes.Buffer)
+		s := fmt.Sprintf("CREATE TABLE IF NOT EXISTS %s (YCSB_KEY VARCHAR(64) PRIMARY KEY", tableName)
+		buf.WriteString(s)
+
+		for i := int64(0); i < fieldCount; i++ {
+			buf.WriteString(fmt.Sprintf(", FIELD%d VARCHAR(%d)", i, fieldLength))
+		}
+
+		buf.WriteString(");")
+
+		_, err := sqlDB.Exec(buf.String())
+		return err
+	}
+	return nil
```

Comment:
Is the init step designed only for sql db?
Maybe the Init step should return an error when sqlDB is nil?

Reply:
1. this init step currently only supports to init ycsb schema on mysql db, other dbs' are still created when DB.Create, so, currently this function return nil.
2. for the apis between workload and db interface, decoupling need to be done. and this PR is only for mysql DB, the ToSqlDB() api is also not suitable here. all these jobs will be done in next PR.